src/v/kafka/server/handlers/fetch.cc

```diff
@@ -89,6 +90,11 @@ static ss::future<read_result> read_from_partition(
 
     reader_config.strict_max_bytes = config.strict_max_bytes;
     auto rdr = co_await part.make_reader(reader_config);
+    auto deferred_close = ss::defer([&rdr = rdr.reader]() mutable {
```

Comment:
I'm a bit uncomfortable that closing happens in the background. Not sure what is the best pattern here and in similar places - maybe something like `try {} catch(...) { co_await rdr.finally(); throw; }` or a `with_reader` helper will be better here?

Reply:
`co_await` is not allowed in catch block

Reply:
Ok, so with coroutines it is even more ugly :) So if the helper is ugly as well, I'm okay with doing it in the background.

Reply:
i will migrate it to future based code, will need more indentation but will close the reader in foreground.